server/server.js

```diff
@@ -56,10 +62,55 @@ Server.prototype.terminate = function(done) {
   });
 
   this.sentry.stop();
-  this.socketServer.close();
+
+  if (this.socketServer) {
+    this.socketServer.close();  
+  }
+  
   Core.Persistence.disconnect(done);
 };
 
+Server.prototype.filter = function filter (filter) {
+  this._filters.push(filter);
```

Comment:
named function expressions (`function foo ()` vs `var foo = function ()`) set the `fn.name` property and show up nicely in stack traces for easier debugging. It also aligns with ES6 ClassElementList syntax, eg
```js
class Foo {
  constructor () { }
  methodA () {}
  methodB () {}
}
```